src/Extensions/Logging/EmptyMachineInformation.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Net;
+using System.Net.Sockets;
+using System.Reflection;
+
+namespace Microsoft.Omex.Extensions.Logging
+{
+	/// <summary>
+	/// Base class for machine information
+	/// </summary>
+	public class EmptyMachineInformation : IMachineInformation
+	{
+		/// <summary>
+		/// Create instance of machine information
+		/// </summary>
+		public EmptyMachineInformation()
+		{
+			MachineName = DefaultEmptyValue;
+			MachineId = DefaultEmptyValue;
+			MachineRole = DefaultEmptyValue;
+			MachineClusterIpAddress = IPAddress.None;
+			MachineCount = 1;
+			MachineCluster = MachineId;
+
+			EnvironmentName = DefaultEmptyValue;
+			DeploymentSlice = DefaultEmptyValue;
+
+			RegionName = DefaultEmptyValue;
+			ServiceName = DefaultEmptyValue;
+			BuildVersion = DefaultEmptyValue;
+
+			IsCanary = false;
+			IsPrivateDeployment = false;
+		}
+
+
+		/// <inheritdoc/>
+		public string MachineId { get; protected set; }
+
+
+		/// <inheritdoc/>
+		public string MachineName { get; protected set; }
+
+
+		/// <inheritdoc/>
+		public string MachineRole { get; protected set; }
+
+
+		/// <inheritdoc/>
+		public IPAddress MachineClusterIpAddress { get; protected set; }
+
+
+		/// <inheritdoc/>
+		public int MachineCount { get; protected set; }
+
+
+		/// <inheritdoc/>
+		public string MachineCluster { get; protected set; }
+
+
+		/// <inheritdoc/>
+		public string EnvironmentName { get; protected set; }
+
+
+		/// <inheritdoc/>
+		public string DeploymentSlice { get; protected set; }
+
+
+		/// <inheritdoc/>
+		public string RegionName { get; protected set; }
+
+
+		/// <inheritdoc/>
+		public string ServiceName { get; protected set; }
+
+
+		/// <inheritdoc/>
+		public string BuildVersion { get; protected set; }
+
+
+		/// <inheritdoc/>
+		public bool IsCanary { get; protected set; }
```

Comment:
Fields have protected setter here because not all of the values available in contraction time, so we need to have a way to update them after.
If we will need to create another constructor it should call this constructor first and then override the required values. Also, I like to have all default initializations in one place, it also helps  to keep in mind that they should be as lightweight as possible

Reply:
Added comment about keeping it lightweight